openedx/core/djangoapps/dark_lang/views.py

```diff
@@ -104,8 +103,6 @@ def _clear_preview_language(self, request):
         Clears the preview language for the current user.
         """
         delete_user_preference(request.user, DARK_LANGUAGE_KEY)
-        if LANGUAGE_SESSION_KEY in request.session:
-            del request.session[LANGUAGE_SESSION_KEY]
```

Comment:
I don't know enough about the transition, but do we need a cookie equivalent? If we find the cookie, should we be expiring it?

Reply:
Is `test_clear_lang` testing this case? I guess I'm still not clear why this doesn't need a replacement.

Reply:
Yes, that case is handling this scenario, also we are now setting/clearing cookies in middleware. We have two middlewares ([LanguagePreferenceMiddleware](https://github.com/openedx/edx-platform/blob/master/openedx/core/djangoapps/lang_pref/middleware.py), [DarkLangMiddleware](https://github.com/openedx/edx-platform/blob/master/openedx/core/djangoapps/dark_lang/middleware.py)) that work cascade order and set/unset cookies on the basis of `user_preferences`. So we already have a step to delete `user_preference` so the rest can be taken care of by the set of middlewares.

And for this particular view and clearing of preview language, DarkLangMiddleware doesn't set the cookie with preview language value, and when nothing is set by DarkLangMiddleware, LanguagePreferenceMiddleware sets the cookie to the user's default preferred language or altogether unset the cookie if no preferred language is found (https://github.com/openedx/edx-platform/blob/master/openedx/core/djangoapps/lang_pref/middleware.py#L92).

Reply:
Also tested this on my devstack as well as on sandbox here: https://languagecookie.sandbox.edx.org/update_lang